ioread.cpp

```diff
@@ -1389,3 +1409,102 @@ bool readFileCells(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
    exitOnError(success,"(READ_FILE_CELLS) Other error",MPI_COMM_WORLD);
    return success;
 }
+
+
+bool readFsgridDecomposition(vlsv::ParallelReader& file, std::array<FsGridTools::Task_t,3>& decomposition){
+   list<pair<string,string> > attribs;
+   uint64_t arraySize;
+   uint64_t vectorSize;
+   vlsv::datatype::type dataType;
+   uint64_t byteSize;
+
+   int myRank;   
```

Comment:
I wonder if it'd be better to perform this with MASTER_RANK only, and then communicate the data? Instead of every task reading numWritingRanks single values from the file at non-contiguous positions. Or will the `file.read`do reading and MPI distribution under the hood?

Reply:
It is piped to the collective MPI IO, but indeed maybe better to be safe and do it from master only.